homeassistant/components/konnected/config_flow.py

```diff
@@ -435,123 +339,278 @@ def pins_to_zones(config):
+            self.data[CONF_HOST] = user_input[CONF_HOST]
+            self.data[CONF_PORT] = user_input[CONF_PORT]
+            self.data[CONF_ACCESS_TOKEN] = self.hass.data.get(DOMAIN, {}).get(
+                CONF_ACCESS_TOKEN
+            ) or "".join(
+                random.choices(f"{string.ascii_uppercase}{string.digits}", k=20)
+            )
 
-        # brief delay to allow processing of recent status req
-        await asyncio.sleep(0.1)
+            try:
+                # brief delay to allow processing of recent status req
+                await asyncio.sleep(0.1)
+                status = await get_status(
+                    self.hass, self.data[CONF_HOST], self.data[CONF_PORT]
+                )
+                self.data[CONF_ID] = status["mac"].replace(":", "")
+                self.data[CONF_MODEL] = status.get("name", KONN_MODEL)
+                return await self.async_step_confirm()
 
-        # try to obtain the mac address from the device
-        try:
-            self.device_id = self.context["device_id"] = (
-                (await get_status(self.hass, self.host, self.port))
-                .get("mac")
-                .replace(":", "")
-            )
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
 
-        except CannotConnect:
-            return self.async_abort(reason="cannot_connect")
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST, default=self.data.get(CONF_HOST)): str,
+                    vol.Required(CONF_PORT, default=self.data.get(CONF_PORT)): int,
+                }
+            ),
+            errors=errors,
+        )
 
-        if not self.device_id:
-            return self.async_abort(reason="cannot_connect")
+    async def async_step_confirm(self, user_input=None):
+        """Attempt to link with the Konnected panel.
 
-        # look for a partially configured instance of this device and hijack it
-        for flow in self.hass.config_entries.flow.async_progress():
-            if flow["handler"] == self.handler and flow["flow_id"] != self.flow_id:
+        Given a configured host, will ask the user to confirm and finalize
+        the connection.
+        """
+        if user_input is None:
+            # update an existing config entry if host info changes
+            entry = await self.async_set_unique_id(
+                self.data[CONF_ID], raise_on_progress=False
+            )
+            if entry and (
+                entry.data[CONF_HOST] != self.data[CONF_HOST]
+                or entry.data[CONF_PORT] != self.data[CONF_PORT]
+            ):
+                entry.data.update(self.data)
+                self.hass.config_entries.async_update_entry(
+                    entry, data=copy.deepcopy(entry.data)
+                )
+
+            # look for a partially imported instance of this device and hijack it
```

Comment:
When does this happen and why do we need to look at other flows than the current? If the import flow was missing host info, it will call the user step so it would still be the same flow here as was started at the import step.

Reply:
This is to handle when ssdp initiates a flow for containing host info for a device that was imported without host info.

Reply:
I think we shouldn't let the ssdp flow help with the import flow. We'll ask the user to complete the import flow anyway.

It's a bad sign that we have to access private attributes to do this.

Reply:
yea - it's dirty feeling. I'm fine removing that function.  It's a definite edge case.